index/index_monitor.go

```diff
@@ -0,0 +1,46 @@
+package index
+
+// helpers to atomically switch from using one index to another
+
+import (
+	"sync"
+)
+
+type IndexMonitor struct {
+	rwMutex sync.RWMutex
+	index *Index
+}
+
+func (im *IndexMonitor) AcquireRef() *Index {
```

Comment:
regardless of what we end up doing here, can you make sure to add comments on all the exported methods?